ansible/roles/debops.docker/tasks/main.yml

```diff
@@ -0,0 +1,55 @@
+---
+- name: Get upstream APT GPG key
+  apt_key:
+    id: '{{ docker__upstream_key }}'
+    keyserver: '{{ ansible_local.core.keyserver
+                   if (ansible_local|d() and ansible_local.core|d() and
+                       ansible_local.core.keyserver)
+                   else "hkp://pool.sks-keyservers.net" }}'
+    state: 'present'
+  register: docker__register_apt_key
+  until: docker__register_apt_key is succeeded
+
+- name: Configure upstream APT repository
+  apt_repository:
+    repo: '{{ docker__upstream_repository }}'
+    mode: '0644'
+    state: 'present'
+    update_cache: True
+
+- name: Install required packages
+  apt:
+    name: '{{ (docker__mandatory_packages
+               + docker__base_packages
+               + docker__packages) | flatten }}'
+    state: 'present'
+    install_recommends: False
+  register: docker__register_packages
+  until: docker__register_packages is succeeded
+
+- name: Ensure /root/.docker/ exists
+  file:
+    path: '/root/.docker'
+    state: 'directory'
+    owner: 'root'
+    group: 'root'
```

Comment:
The role is executed as `root` already, you should drop the `owner` and `group` in such cases. Use them only if you want to change the ownership to a specific account/group.